Update Copy Permalink for GitHub

dcrousso opened the pull request:
#### 48f0074ad44092c125ed23f1748df01b2c107a49
<pre>
Update Copy Permalink for GitHub
<a href="https://bugs.webkit.org/show_bug.cgi?id=242166">https://bugs.webkit.org/show_bug.cgi?id=242166</a>

Reviewed by Jonathan Bedard.

* Tools/CopyPermalink/Sublime Text/CopyWebKitPermalink/CopyWebKitPermalink.py:
(Settings.match_commit): Renamed from `include_revision`.
(CopyWebKitPermalinkCommand.run):
(CopyWebKitPermalinkCommand.is_git_directory):
(CopyWebKitPermalinkCommand.path_relative_to_repository_root_for_path):
(CopyWebKitPermalinkCommand.current_git_branch):
(CopyWebKitPermalinkCommand.current_git_commit):
(CopyWebKitPermalinkCommand.determine_vcs_from_path): Deleted.
(CopyWebKitPermalinkCommand.path_is_in_webkit_checkout): Deleted.
(CopyWebKitPermalinkCommand.git_path_relative_to_repository_root_for_path): Deleted.
(CopyWebKitPermalinkCommand.svn_path_relative_to_repository_root_for_path): Deleted.
(CopyWebKitPermalinkCommand.revision_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.svn_revision_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.git_revision_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.svn_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.permalink_for_path): Deleted.
(CopyWebKitPermalinkCommand.is_svn_directory): Deleted.
(CopyWebKitPermalinkCommand.is_git_svn_directory): Deleted.
* Tools/CopyPermalink/Sublime Text/CopyWebKitPermalink/CopyWebKitPermalink.sublime-settings:

Canonical link: <a href="https://commits.webkit.org/252051@main">https://commits.webkit.org/252051@main</a>
</pre>

JonWBedard commented on Tools/CopyPermalink/Sublime Text/CopyWebKitPermalink/CopyWebKitPermalink.py:
```diff
@@ -80,19 +81,21 @@ def run(self, edit, annotate_blame=False):
-        path = self.path_relative_to_repository_root_for_path(document_path)
-        revision_info = self.revision_info_for_path(document_path)
-        permalink = self.permalink_for_path(path, line_number, revision_info, annotate_blame)
+        permalink = 'https://github.com/WebKit/WebKit/{}/{}/{}#L{}'.format(mode, branch, urllib.parse.quote(path), line_number)
```
Any reason we're restricting this to WebKit? It should generally work for any GitHub project, yes?

JonWBedard reviewed (commented):
I'm not familiar with this tool, is there any documentation for what we're trying to do at a high-level here?

dcrousso committed:
Update Copy Permalink for GitHub
https://bugs.webkit.org/show_bug.cgi?id=242166

Reviewed by Jonathan Bedard.

* Tools/CopyPermalink/Sublime Text/CopyWebKitPermalink/CopyWebKitPermalink.py:
(Settings.match_commit): Renamed from `include_revision`.
(CopyWebKitPermalinkCommand.run):
(CopyWebKitPermalinkCommand.is_git_directory):
(CopyWebKitPermalinkCommand.path_relative_to_repository_root_for_path):
(CopyWebKitPermalinkCommand.current_git_branch):
(CopyWebKitPermalinkCommand.current_git_commit):
(CopyWebKitPermalinkCommand.determine_vcs_from_path): Deleted.
(CopyWebKitPermalinkCommand.path_is_in_webkit_checkout): Deleted.
(CopyWebKitPermalinkCommand.git_path_relative_to_repository_root_for_path): Deleted.
(CopyWebKitPermalinkCommand.svn_path_relative_to_repository_root_for_path): Deleted.
(CopyWebKitPermalinkCommand.revision_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.svn_revision_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.git_revision_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.svn_info_for_path): Deleted.
(CopyWebKitPermalinkCommand.permalink_for_path): Deleted.
(CopyWebKitPermalinkCommand.is_svn_directory): Deleted.
(CopyWebKitPermalinkCommand.is_git_svn_directory): Deleted.
* Tools/CopyPermalink/Sublime Text/CopyWebKitPermalink/CopyWebKitPermalink.sublime-settings:

Canonical link: https://commits.webkit.org/252051@main
